jproperties-checks/src/main/java/org/sonar/jproperties/checks/KeyNamingConventionCheck.java

```diff
@@ -42,7 +42,7 @@
 @ActivatedByDefault
 public class KeyNamingConventionCheck extends SquidCheck<LexerlessGrammar> {
 
-  private static final String DEFAULT_FORMAT = "^[a-z]+[-._a-z0-9]*$";
+  private static final String DEFAULT_FORMAT = "^[.a-zA-Z0-9]+$";
```

Comment:
I don't think that allowing numbers, dot and uppercase character at the beginning of name is good practice - it should be rather `[a-z][.a-zA-Z0-9]+`. Also, we could think of marking as invalid propery like `a.........=something`